homework_weekend7/src/task1/LinkedListBoolean.java

```diff
@@ -0,0 +1,198 @@
+package task1;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Represents implementation of LinkedList for boolean values. Implements common
+ * methods related to the LinkedList.
+ * 
+ * @author ajla
+ *
+ */
+
+public class LinkedListBoolean {
+	private Node head;
+
+	/**
+	 * Adds element at the end of the list.
+	 * 
+	 * @param value
+	 * @return
+	 */
+	public void add(boolean value) {
+		if (head == null) {
+			head = new Node(value);
+			return;
+		}
+
+		Node tmp = head;
+
+		while (tmp.getNextNode() != null) {
+			tmp = tmp.getNextNode();
+		}
+		tmp.setNextNode(new Node(value));
+	}
+
+	/**
+	 * Removes first element with given value.
+	 * 
+	 * @param elem
+	 * @throws NoSuchElementException
+	 */
+	public void remove(boolean elem) throws NoSuchElementException {
+		if (head == null)
+			throw new NoSuchElementException();
+		if (head.getValue() == elem) {
+			head = head.getNextNode();
+			return;
+		}
+
+		Node tmp = head.getNextNode();
+		while (tmp != null) {
+			if (tmp.next.getValue() == elem) {
+				tmp.setNextNode(tmp.next.next);
+				break;
+			} else {
+				tmp = tmp.next;
+			}
+		}
+		throw new NoSuchElementException();
+	}
+
+	/**
+	 * Removes first element with given value, after given index.
+	 * 
+	 * @param elem
+	 * @param index
+	 * @throws NoSuchElementException
+	 * @throws ArrayIndexOutOfBoundsException
+	 */
+	public void remove(boolean elem, int index) throws NoSuchElementException,
+			ArrayIndexOutOfBoundsException {
+		int size = size(), count = 0;
+		Node tmp = head;
+
+		for (int i = 0; i < size; i++) {
+			if (head == null || index < 0 || index >= size) {
```

Comment:
Index validation should be moved out from the loop. It could be performed right after you determined the size of the list.